rules/framework/vfs_overlay.bzl

```diff
@@ -64,20 +70,31 @@ def _make_root(vfs_parent, bin_dir_path, build_file_path, framework_name, swiftm
         })
 
     if len(swiftmodules):
-        # Assuming each swift module content share the same parent
-        parent_dir_base_name = swiftmodules[0].dirname.split("/").pop()
-        modules_contents.append({
-            "type": "directory",
-            "name": parent_dir_base_name,
-            "contents": [
-                {
-                    "type": "file",
-                    "name": file.basename,
-                    "external-contents": _get_external_contents(vfs_prefix, file.path),
-                }
-                for file in swiftmodules
-            ],
-        })
+        any_swiftmodule_file = swiftmodules[0]
+
+        # Virtualize a swiftmodules directory inside of the framework only for
+        # user provided source swiftmodules nested in a directory.
+        #
+        # Note: This code has some edges: perhaps for imported/source
+        # frameworks it should mirror the users's structure directly. See the
+        # related PR to import that source.
+        if any_swiftmodule_file.is_source:
+            # Handle a glob of files inside of a .swiftmodule e.g. for xcframework
+            parent_dir_base_name = any_swiftmodule_file.dirname.split("/").pop()
+            swiftmodule_file = _find_top_swiftmodule_file(swiftmodules)
+            if not swiftmodule_file and parent_dir_base_name.endswith(".swiftmodule"):
```

Comment:
so if this is not true, aka we have a .swiftmodule file as a source file not dir, and parent dir is not ending with .swiftmodule, we don't append the content to module_content.. 
on the other hand (code below) if we have a list of swiftmodules, it does the other side of logic... (return None if...)
so these two blocks of code actually have some links between each other?

Reply:
In this PR ( https://github.com/bazel-ios/rules_ios/pull/280 ) it started added the parent build file's directory, so that would put unexpected entires in VFS: e.g. if you had a parent of `Public` the could would add a `Public`  add that in the modules directory in the VFS. 

The code makes it more explicit to handle or other issues

Reply:
discussed and we think it's possible to merge this with code down below. Will deal with this in another PR but for now all these makes sense